tests/functional/test_foreign_references.py

```diff
@@ -0,0 +1,220 @@
+    )
+)
+
+test_db_name = str(os.environ.get("TEST_DB_NAME", "test_db2"))
+test_connection_string = str(
+    os.environ.get(
+        "TEST_DB_CONNECTION_STRING",
+        f"postgresql://postgres:{db_user_password}@localhost:5432/{test_db_name}?sslmode=disable",
+    )
+)
+testdb = DBInfo(DatabaseType.aiopg_engine, test_connection_string)
+
+
+@database_model("organizations", testdb)
+class Org(BaseModel):
+    """A test class for testing. Linter won't let me not have this useless comment :)"""
+
+    id: UUID_STR = Field(primary_key=True)
+    serial_id: Optional[int]
+    name: str = Field(max_length=45)
+    slug: str = Field(max_length=45)
+
+    created_at: Optional[datetime] = Field(fetch_on_create=True)
+    updated_at: Optional[datetime] = Field(fetch_on_update=True)
+
+
+@database_model("apps", testdb)
+class App(BaseModel):
+    """Model that represents the App table schema"""
+
+    id: Optional[UUID_STR] = Field(primary_key=True, fetch_on_create=True)
+    name: str = Field(max_length=45)
+    org: foreign_key(Org, reference_field="organization_id")
+    fiscal_year_start_month: Optional[int]
+    metadata: Optional[dict]
+    slug: str = Field(max_length=45)
+    data_cache_schema: str
+    data_store_schema: str
+    serial_id: Optional[int] = Field(fetch_on_create=True)
+    created_at: Optional[datetime] = Field(fetch_on_create=True)
+    updated_at: Optional[datetime] = Field(fetch_on_update=True)
+    apple_compliant: Optional[bool] = Field(default=False)
+    deleted: Optional[bool] = Field(default=False)
+
+
+@database_model("topics", testdb)
+class Topic(BaseModel):
+    """A test class with a nullable JSONB field"""
+
+    id: UUID_STR = Field(primary_key=True)
+    app: foreign_key(App, reference_field="app_id")
+    filter: Optional[dict] = Field()
+    filter_hash: str = Field(max_length=45)
+    name: str = Field(max_length=45)
+
+
+async def setup_db_and_tables():
+    """Create the database and tables"""
+    await create_new_database(existing_connection_string, test_db_name)
+
+    await create_table(testdb, Org._table)
+    await create_table(testdb, Topic._table)
+    await create_table(testdb, App._table)
+    conn = await (await DBEngine.get_engine(testdb)).acquire()
+    await add_trigger(conn, "organizations")
+
+
+loop = asyncio.get_event_loop()
+loop.run_until_complete(setup_db_and_tables())
+
+
+@pytest.mark.asyncio
+async def test_fetch() -> None:
+    """Test that we can resolve the reference for a foreign key"""
+    org_id = str(uuid4())
+    serial_id = 104
+
+    try:
+        org = await Org.create(
+            id=org_id, serial_id=serial_id, name="fake org104", slug="fake slug104"
+        )
+        app = await App.create(
+            id=str(uuid4()),
+            name="app name",
+            org=org,
+            slug="app-slug",
+            data_cache_schema="app_ds1",
+            data_store_schema="app_dc1",
+        )
+
+        app_get = await App.get(app.id)
```

Comment:
This is probably just me missing something -- you aren't including `resolve_references=True` in this case, so this is like, an org object that is empty except for the id? And it is only after you call fetch on the org that the other stuff gets filled in?

Reply:
correct, if you try to access any org properties except for its id it will throw an error. Once you call `fetch()` it will be fully resolved.

Im going to the update the error to a specific one instead of `AttributeError`